File: project/donghyeon-onboarding/src/main/java/me/dhlee/donghyeononboarding/domain/Survey.java
package me.dhlee.donghyeononboarding.domain;

import org.hibernate.annotations.SQLRestriction;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.dhlee.donghyeononboarding.common.BaseEntity;
import me.dhlee.donghyeononboarding.exception.AppException;
import me.dhlee.donghyeononboarding.exception.ErrorCode;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@SQLRestriction("deleted_at is null")
@Table(name = "Surveys")
@Entity
public class Survey extends BaseEntity {

    @Column(length = 100, nullable = false)
    private String title;

    @Column(nullable = false)
    private String description;

    @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<SurveyItem> items = new HashSet<>();

    @Builder
    private Survey(String title, String description) {
        this.title = title;
        this.description = description;
    }

    public void addItem(SurveyItem item) {
        if (isFull()) {
            throw new AppException(ErrorCode.SURVEY_ITEM_SIZE_OVERFLOW);
        }
        this.items.add(item);
    }

    private boolean isFull() {
        return items.size() >= 10;
    }
}

File: project/seonghun-onboarding/src/main/kotlin/com/chosseang/seonghunonboarding/entity/Survey.kt
package com.chosseang.seonghunonboarding.entity

import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany

@Entity
data class Survey(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    val name: String?,
    val description: String,

    @OneToMany(mappedBy = "survey", orphanRemoval = true, cascade = [CascadeType.ALL])
    var items: MutableList<Item>
){
    // 아이템 추가 메서드 - 양방향 관계 유지
    fun addItem(item: Item) {
        items.add(item)
        item.survey = this
    }

    // 여러 아이템 한번에 추가
    fun addItems(itemList: List<Item>) {
        itemList.forEach { addItem(it) }
    }
}
